src/commands/git-metadata/gitdb.ts

```diff
@@ -0,0 +1,281 @@
+  } catch (err) {
+    log.warn(`Failed getting commits to exclude: ${err}`)
+    throw err
+  }
+
+  let commitsToUpload
+  try {
+    commitsToUpload = await getCommitsToUpload(git, commitsToExclude)
+    log.debug(`${commitsToUpload.length} commits to upload.`)
+  } catch (err) {
+    log.warn(`Failed getting commits to upload: ${err}`)
+    throw err
+  }
+
+  let packfiles
+  try {
+    packfiles = generatePackFilesForCommits(log, commitsToUpload)
+    log.debug(`${packfiles.length} packfiles generated.`)
+  } catch (err) {
+    log.warn(`Failed generating packfiles: ${err}`)
+    throw err
+  }
+
+  if (dryRun) {
+    log.debug(`Dry-run enabled, not uploading anything.`)
+
+    return
+  }
+  log.debug(`Uploading packfiles...`)
+  try {
+    await uploadPackfiles(log, request, repoURL, latestCommits[0], packfiles)
+    log.debug(`Successfully uploaded packfiles.`)
+  } catch (err) {
+    log.warn(`Failed to upload packfiles: ${err}`)
+    throw err
+  }
+}
+
+const getRepoURL = gitRemote
+
+const getLatestLocalCommits = async (git: simpleGit.SimpleGit) => {
+  const logResult = await git.log(['-n 1000', '--since="1 month ago"'])
+
+  return logResult.all.map((c) => c.hash)
+}
+
+// getKnownCommits asks the backend which of the given commits are already known
+const getKnownCommits = async (log: Logger, request: RequestBuilder, repoURL: string, latestCommits: string[]) => {
+  interface SearchCommitResponse {
+    data: Commit[]
+  }
+
+  interface Commit {
+    type: string
+    id: string
+  }
+
+  const localCommitData = JSON.stringify({
+    meta: {
+      repository_url: repoURL,
+    },
+    data: latestCommits.map((commit) => ({
+      id: commit,
+      type: 'commit',
+    })),
+  })
+  const response = await runRequest(log, 'search_commits', () =>
+    request({
+      url: '/api/v2/git/repository/search_commits',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      data: localCommitData,
+      method: 'POST',
+      timeout: API_TIMEOUT,
+    })
+  )
+  const commits = response.data as SearchCommitResponse
+  if (!commits || commits.data === undefined) {
+    throw new Error(`Invalid API response: ${response}`)
+  }
+
+  return commits.data.map((c) => {
+    if (c.type !== 'commit' || c.id === undefined) {
+      throw new Error('Invalid commit type response')
+    }
+
+    return sanitizeCommit(c.id)
+  })
+}
+
+const sanitizeCommit = (sha: string) => {
+  const isValidSha = (s: string) => /[0-9a-f]{40}/.test(s)
+
+  const sanitizedCommit = sha.replace(/[^0-9a-f]+/g, '')
+  if (sanitizedCommit !== sha) {
+    throw new Error(`Invalid commit format: ${sha} (different from sanitized ${sanitizedCommit})`)
+  }
+  if (!isValidSha(sanitizedCommit)) {
+    throw new Error(`Invalid commit format: ${sanitizedCommit}`)
+  }
+
+  return sanitizedCommit
+}
+
+const getCommitsToUpload = async (git: simpleGit.SimpleGit, commitsToExclude: string[]) => {
```

Comment:
This function will return ids for commits and trees, so I would use another name.

```suggestion
const getObjectsToUpload = async (git: simpleGit.SimpleGit, commitsToExclude: string[]) => {
```